File: robocode-tankroyale-bot-api-java/src/main/java/dev/robocode/tankroyale/botapi/events/BotHitWallEvent.java
package dev.robocode.tankroyale.botapi.events;

/** Event occurring when the bot has hit a wall. */
@SuppressWarnings("unused")
public final class BotHitWallEvent extends Event {

  /** ID of the victim bot that hit the wall. */
  private final int victimId;

  /**
   * Initializes a new instance of the BotHitWallEvent class.
   *
   * @param turnNumber is the turn number when the bot has hit the wall.
   * @param victimId is the ID of the victim bot that hit the wall.
   */
  public BotHitWallEvent(int turnNumber, int victimId) {
    super(turnNumber);
    this.victimId = victimId;
  }
  /**
   * Returns the ID of the victim bot that hit the wall.
   *
   * @return The ID of the victim bot that hit the wall.
   */
  public int getVictimId() {
    return victimId;
  }
}

File: server/src/main/kotlin/dev/robocode/tankroyale/server/event/BotHitWallEvent.kt
package dev.robocode.tankroyale.server.event

import dev.robocode.tankroyale.server.model.BotId

/** Event sent when a bot has hit a wall. */
class BotHitWallEvent(
    /** Turn number when event occurred */
    override val turnNumber: Int,

    /** Bot id of the victim that has hit a wall */
    val victimId: BotId,
) : Event()